src/main/java/com/google/devtools/build/lib/actions/ActionCacheChecker.java

```diff
@@ -287,6 +407,12 @@ public Token getTokenIfNeedToExecute(
       }
     }
     ActionCache.Entry entry = getCacheEntry(action);
+    CachedOutputMetadata cachedOutputMetadata = null;
+    // load remote metadata from action cache
+    if (entry != null && !entry.isCorrupted() && cacheConfig.storeOutputMetadata()) {
+      cachedOutputMetadata = loadRemoteOutputMetadataIfMissing(action, entry, metadataHandler);
```

Comment:
Nice! I like that we skip the load if that is disabled. Allows for a good feature switch.